src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexWriteOperation.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Corax;
+using Raven.Client.Documents.Indexes;
+using Raven.Server.Exceptions;
+using Raven.Server.Utils;
+using Sparrow.Json;
+using Sparrow.Logging;
+using Voron;
+using Voron.Impl;
+
+namespace Raven.Server.Documents.Indexes.Persistence.Corax
+{
+    public class CoraxIndexWriteOperation : IndexWriteOperationBase
+    {
+        private readonly IndexWriter _indexWriter;
+        private readonly CoraxDocumentConverter _converter;
+        private readonly Dictionary<Slice, int> _knownFields;
+        private int _entriesCount = 0;
+
+        public CoraxIndexWriteOperation(Index index, Transaction writeTransaction, CoraxDocumentConverter converter, Logger logger) : base(index, logger)
+        {
+            _converter = converter;
+            _knownFields = _converter.GetKnownFields();
+            try
+            { 
+                _indexWriter = new IndexWriter(writeTransaction);
+            }
+            catch (Exception e) when (e.IsOutOfMemory())
+            {
+                throw;
+            }
+            catch (Exception e)
+            {
+                throw new IndexWriteException(e);
+            }
+        }
+
+        public override void Dispose()
+        {
+            _indexWriter?.Dispose();
+        }
+
+        public override void Commit(IndexingStatsScope stats)
+        {
+            if (_indexWriter != null)
+            {
+                using (stats.For(IndexingOperation.Corax.Commit))
+                    _indexWriter.Commit();
+            }
+        }
+
+        public override void IndexDocument(LazyStringValue key, LazyStringValue sourceDocumentId, object document, IndexingStatsScope stats, JsonOperationContext indexContext)
+        {
+            EnsureValidStats(stats);
+            _entriesCount++;
+            Span<byte> data;
+            LazyStringValue lowerId;
+
+            using (Stats.ConvertStats.Start())
+                data = _converter.InsertDocumentFields(key, sourceDocumentId, document, indexContext, out lowerId);
+
+            _indexWriter.Index(lowerId, data, _knownFields);
```

Comment:
Don't we need stats here?

Reply:
We are measuring the time of preparing Doc for indexing.
@ppekrol ?

Reply:
You are measuring conversions results, but not `AddToIndex` time as well

Reply:
Agreed, added writing stats.